### PR TITLE
WIP,MNT: disable 3d options during testing

### PR DESCRIPTION
**Pull request by GuillaumeFavelier**
This PR follows https://github.com/mne-tools/mne-python/pull/10319#discussion_r804679165 and introduces a `pytest.fixture` called `options_3d` to disable advanced 3d rendering options.

I tried setting `scope='session'` but I got the following error:

```
_______________________________________________________________ ERROR at setup of test_3d_functions[pyvistaqt] _______________________________________________________________
ScopeMismatch: You tried to access the 'function' scoped fixture 'monkeypatch' with a 'session' scoped request object, involved factories
mne/conftest.py:655:  def options_3d(monkeypatch)
../anaconda3/envs/mne-py38/lib/python3.8/site-packages/_pytest/monkeypatch.py:29:  def monkeypatch() -> Generator[ForwardRef('MonkeyPatch'), NoneType, NoneType]
```

cc @larsoner

### CODE REVIEW

**Comment by GuillaumeFavelier** [comment]
And I think it really needs to be `scope='session'` because GUI uses it too:

```
==================================== ERRORS ====================================
_________________ ERROR at setup of test_gui_api_qt[pyvistaqt] _________________
ScopeMismatch: You tried to access the function scoped fixture options_3d with a module scoped request object, involved factories:
mne/conftest.py:462:  def renderer_interactive_pyvistaqt(request, options_3d)
mne/conftest.py:655:  def options_3d(monkeypatch)
```

**Comment by GuillaumeFavelier** [comment]
Probably, an idea for a follow-up PR would be:

- add a 3d option for smooth shading (to be disabled in testing in the same way)

**Review by agramfort — APPROVED** [review]

**Comment by larsoner** [comment]
I would just make all the renderer fixtures use this fixture, then, if it has to be function level

File: mne/conftest.py [hunk 1]
```diff
@@ -652,6 +653,18 @@ def download_is_error(monkeypatch):
...
+@pytest.fixture(scope='module')
+def options_3d():
+    """Disable advanced 3d rendering."""
+    with mock.patch.dict(
+        os.environ, {
+            "MNE_3D_OPTION_ANTIALIAS": "false",
+            "MNE_3D_OPTION_DEPTH_PEELING": "false",
+        }
+    ):
+        yield
```
[the remark this answers is not in the document]
Comment (larsoner) [reply mne/conftest.py:665]: (might need `scope='function'` above, which is fine to change it to if it's the case)

**Comment by GuillaumeFavelier** [comment]
The current solution ([765721b](https://github.com/mne-tools/mne-python/pull/10334/commits/765721b76d02bcacc7f0608348e53c1c05b8b230))  is not acceptable @larsoner ?

---

I tried your suggestion while changing the scope (`function`, `module` or `session`) but I get a `ScopeMismatch` in all cases:

`function`:

```
=================================================================================== ERRORS ===================================================================================
________________________________________________________________ ERROR at setup of test_gui_api_qt[pyvistaqt] ________________________________________________________________
ScopeMismatch: You tried to access the 'function' scoped fixture 'options_3d' with a 'module' scoped request object, involved factories
mne/conftest.py:462:  def renderer_interactive_pyvistaqt(request, options_3d)
mne/conftest.py:655:  def options_3d(monkeypatch)
```

---

`module`:

```
=================================================================================== ERRORS ===================================================================================
__________________________________________________________________ ERROR at setup of test_gui_api[notebook] __________________________________________________________________
ScopeMismatch: You tried to access the 'function' scoped fixture 'monkeypatch' with a 'module' scoped request object, involved factories
mne/conftest.py:655:  def options_3d(monkeypatch)
../anaconda3/envs/mne-py38/lib/python3.8/site-packages/_pytest/monkeypatch.py:29:  def monkeypatch() -> Generator[ForwardRef('MonkeyPatch'), NoneType, NoneType]
________________________________________________________________ ERROR at setup of test_gui_api_qt[pyvistaqt] ________________________________________________________________
ScopeMismatch: You tried to access the 'function' scoped fixture 'monkeypatch' with a 'module' scoped request object, involved factories
mne/conftest.py:462:  def renderer_interactive_pyvistaqt(request, options_3d)
mne/conftest.py:655:  def options_3d(monkeypatch)
```

`session`:

```
__________________________________________________________________ ERROR at setup of test_gui_api[notebook] __________________________________________________________________
ScopeMismatch: You tried to access the 'function' scoped fixture 'monkeypatch' with a 'session' scoped request object, involved factories
mne/conftest.py:655:  def options_3d(monkeypatch)
../anaconda3/envs/mne-py38/lib/python3.8/site-packages/_pytest/monkeypatch.py:29:  def monkeypatch() -> Generator[ForwardRef('MonkeyPatch'), NoneType, NoneType]
________________________________________________________________ ERROR at setup of test_gui_api_qt[pyvistaqt] ________________________________________________________________
ScopeMismatch: You tried to access the 'function' scoped fixture 'monkeypatch' with a 'session' scoped request object, involved factories
mne/conftest.py:655:  def options_3d(monkeypatch)
```

I'm no expert at monkeypatching and it could be something wrong with my configuration so I'll push anyway.

**Comment by larsoner** [comment]
Sorry for the runaround @GuillaumeFavelier I guess it's not possible :(

I reverted back to the `unittest.mock` approach and then deprecated `modified_env` since we shouldn't need it anymore. Will merge once CIs come back happy!

**Pull request merged**